tests/contrib/starlette/app.py

```diff
@@ -52,7 +67,56 @@ async def file(request):
         return FileResponse(fp.name)
 
 
-def get_app():
+async def get_tables(request):
+    global engine
+    response = engine.table_names()
+    return PlainTextResponse(str(response))
+
+
+async def list_notes(request):
+    global engine
+    query = "SELECT * FROM NOTES"
+    with engine.connect() as connection:
+        result = connection.execute(query)
+        d, a = {}, []
+        for rowproxy in result:
+            for column, value in rowproxy.items():
+                d = {**d, **{column: value}}
+            a.append(d)
+    response = str(a)
+    return PlainTextResponse(response)
+
+
+async def add_note(request):
+    global engine
+    global notes_table
+    request_json = await request.json()
+    with engine.connect() as connection:
+        with connection.begin():
+            connection.execute(notes_table.insert(), request_json)
+    response = "Success"
+    return PlainTextResponse(response)
+
+
+def get_app(test_engine):
+    global engine
+    engine = test_engine
+
+    create_test_database(engine)
+
+    DATABASE_URL = "sqlite:///test.db"
+    database = databases.Database(DATABASE_URL)
+    metadata = sqlalchemy.MetaData(bind=engine)
+    metadata.reflect()
+
+    global notes_table
+    notes_table = metadata.tables["notes"]
+
+    row = {"id": 1, "text": "test", "completed": 1}
```

Comment:
This shouldn't be done in `get_app`. And by removing this I think you can get rid of all the global notes_table stuff.